### PR TITLE
Initialize OC in genesis

### LINKED ISSUE #221: Initialize OC in genesis
After discussion with Martin, we decided that for both OC (Oversight Community) and AP (Arbiter Pool), we will already know off-chain the identities of the genesis participants and have their consent for initial setup. This means we can take a list of members in the genesis file and set it up during `InitGenesis` call. The process will be:

Validate:
* Ensure all accounts in the initial list have a bank balance of more than "escrow_amount" `utgd`

Execute:

* Create Trusted-Circle (OC or AP) using the first member of the member list as creator, and sending the min escrow. (Replace "systemAdmin" as creator [2nd param](https://github.com/confio/tgrade/blob/v0.6.2/x/poe/bootstrap.go#L92))
* Creator makes a proposal to add **all other members** ( example for a gov [proposal](https://github.com/confio/tgrade/blob/v0.6.2/x/poe/contract/tgrade_oc_proposals_integration_test.go#L45) )
* Create executes this proposal  ([example](https://github.com/confio/tgrade/blob/v0.6.2/x/poe/contract/tgrade_oc_proposals_integration_test.go#L54))
* Loop over all other members, call `DepositEscrow` with the minimum amount
* Check the voting member set is equal to the list of all members defined in genesis (Example: https://github.com/confio/tgrade/blob/v0.6.2/x/poe/contract/tgrade_oc_proposals_integration_test.go#L31))

All these calls can be done in Go and is the only way to send tokens from many accounts to a contract.
Edge case to test:
- OC member has not enough tokens for deposit
- Ensure number of members is >0

### PR DESCRIPTION
**Pull request by pinosu**
Resolves #221

### CODE REVIEW

File: x/poe/bootstrap.go [hunk 1]
```diff
@@ -89,14 +89,68 @@ func bootstrapPoEContracts(ctx sdk.Context, k wasmtypes.ContractOpsKeeper, tk tw
...
+		}
+	}
+
+	voters, err := tcAdapter.QueryListVoters(ctx)
+	if err != nil {
+		return sdkerrors.Wrap(err, "query voters")
+	}
+
+	//verify the members list are all voting members
+	notVotingMembers := make(map[string]struct{}, len(ocMembers))
```
Comment (alpe) [inline x/poe/bootstrap.go:142]: Looking at this, I am not sure if we need to have a sanity check at all. We can expect the contract to work proper and just verify the member list in an integration test.

**Review by alpe — COMMENTED** [review]
Very nice start! Bindings and logic look 🏄

File: x/poe/bootstrap.go [hunk 2]
```diff
@@ -89,14 +89,52 @@ func bootstrapPoEContracts(ctx sdk.Context, k wasmtypes.ContractOpsKeeper, tk tw
 	}
 	ocInitMsg := newOCInitMsg(gs)
 	deposit := sdk.NewCoins(gs.OversightCommitteeContractConfig.EscrowAmount)
-	ocContractAddr, _, err := k.Instantiate(ctx, ocCodeID, systemAdminAddr, systemAdminAddr, mustMarshalJson(ocInitMsg), "oversight_committee", deposit)
+
+	ocMembers := make([]string, len(gs.Engagement))
```
Comment (alpe) [inline x/poe/bootstrap.go:93]: The OversightCommunity (OC) is not the same as the engagement addresses. Although there can be some overlap. 
We would need a new element in genesis to define the addresses for all OC members.

File: x/poe/bootstrap.go [hunk 3]
```diff
@@ -91,14 +91,48 @@ func bootstrapPoEContracts(ctx sdk.Context, k wasmtypes.ContractOpsKeeper, tk tw
...
+		}
+		// deposit escrow
+		for _, member := range gs.OversightCommunityMembers[1:] {
+			ocMember, err := sdk.AccAddressFromBech32(member)
+			if err != nil {
+				return sdkerrors.Wrapf(err, "%s oc member", member)
+			}
+			err = tcAdapter.DepositEscrow(ctx, gs.OversightCommitteeContractConfig.EscrowAmount, ocMember)
+			if err != nil {
+				return sdkerrors.Wrap(err, "deposit escrow")
```
Comment (alpe) [inline x/poe/bootstrap.go:131]: nit: please add the address to the error msg as well. If there are a lot this will help finding the issue.

File: x/poe/bootstrap_integration_test.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+	//verify the members list are all voting members
+	notVotingMembers := make(map[string]struct{}, len(members))
+	for _, m := range members {
+		notVotingMembers[m] = struct{}{}
+	}
+
+	for _, v := range voters.Members {
+		delete(notVotingMembers, v.Addr)
+	}
+	assert.Empty(t, notVotingMembers)
```
Comment (alpe) [inline x/poe/bootstrap_integration_test.go:45]: I got confused with the "notVotingMembers" here. But this test ensures that all OC members are voting members.

File: x/poe/contract/common_test.go [hunk 5]
```diff
@@ -33,6 +33,11 @@ func setupPoEContracts(t *testing.T, mutators ...func(m *types.GenesisState)) (s
 	gs := types.GenesisStateFixture(append([]func(m *types.GenesisState){mutator}, mutators...)...)
 	adminAddress, _ := sdk.AccAddressFromBech32(gs.SystemAdminAddress)
 	example.Faucet.Fund(ctx, adminAddress, sdk.NewCoin(types.DefaultBondDenom, sdk.NewInt(100_000_000_000)))
+	for _, member := range gs.OversightCommunityMembers {
+		addr, err := sdk.AccAddressFromBech32(member)
+		require.NoError(t, err)
+		example.Faucet.Fund(ctx, addr, sdk.NewCoin(types.DefaultBondDenom, sdk.NewInt(1_000_000)))
```
Comment (alpe) [inline x/poe/contract/common_test.go:39]: 👍  some tokens are required

File: x/poe/contract/tgrade_oc_proposals_integration_test.go [hunk 6]
```diff
@@ -14,6 +14,7 @@ import (
 )
 
 func TestSlashValidator(t *testing.T) {
+	t.Skip()
```
Comment (alpe) [inline x/poe/contract/tgrade_oc_proposals_integration_test.go:17]: 👍  As discussed either I take care of this or we can have a tracking issue to not forget it.

File: x/poe/types/test_fixtures.go [hunk 7]
```diff
@@ -40,6 +40,8 @@ func GenesisStateFixture(mutators ...func(m *GenesisState)) GenesisState {
 		Points:  10,
 	}}
 
+	r.OversightCommunityMembers = []string{RandomAccAddress().String(), RandomAccAddress().String()}
```
Comment (alpe) [inline x/poe/types/test_fixtures.go:43]: 👍

**Review by alpe — COMMENTED** [review]
Nice work. Looks very good already

**Review by alpe — APPROVED** [review]
LGTM 🏆
